src/Perspectives/immutable/ungoverned/cluster/ClusterConservativeWithRecognizedCollateralsPerspective.sol

```diff
@@ -0,0 +1,145 @@
+        address oracle = IEVault(vault).oracle();
+        testProperty(routerFactory.isValidDeployment(oracle), ERROR__ORACLE);
+        testProperty(IEulerRouter(oracle).governor() == address(0), ERROR__ORACLE);
+
+        // Verify the unit of account is either USD or WETH
+        address unitOfAccount = IEVault(vault).unitOfAccount();
+        testProperty(unitOfAccount == USD || unitOfAccount == ETH || unitOfAccount == WETH, ERROR__UNIT_OF_ACCOUNT);
+
+        // the router must contain a valid pricing configuration
+        address fallbackOracle = IEulerRouter(oracle).fallbackOracle();
+        {
+            testProperty(fallbackOracle == address(0) || routerFactory.isValidDeployment(fallbackOracle), ERROR__ORACLE);
+            (,,, address resolvedOracle) = IEulerRouter(oracle).resolveOracle(1e18, vault, unitOfAccount);
+            testProperty(
+                (fallbackOracle != address(0) && resolvedOracle == fallbackOracle)
+                    || adapterRegistry.isValidAdapter(resolvedOracle, block.timestamp),
+                ERROR__ORACLE_ASSET
+            );
+        }
+
+        // cluster vaults must have collaterals set up
+        address[] memory ltvList = IEVault(vault).LTVList();
+        testProperty(ltvList.length > 0 && ltvList.length <= 10, ERROR__LTV_LENGTH);
+
+        // cluster vaults must have recognized collaterals with LTV set in range
+        for (uint256 i = 0; i < ltvList.length; ++i) {
+            address collateral = ltvList[i];
+            {
+                (,,, address resolvedOracle) = IEulerRouter(oracle).resolveOracle(1e18, vault, unitOfAccount);
+                testProperty(
+                    (fallbackOracle != address(0) && resolvedOracle == fallbackOracle)
+                        || adapterRegistry.isValidAdapter(resolvedOracle, block.timestamp),
+                    ERROR__ORACLE_COLLATERAL
+                );
+            }
+
+            // cluster vaults must have liquidation discount in a certain range
+            uint16 maxLiquidationDiscount = IEVault(vault).maxLiquidationDiscount();
+            testProperty(
+                maxLiquidationDiscount >= 0.05e4 && maxLiquidationDiscount <= 0.2e4, ERROR__LIQUIDATION_DISCOUNT
+            );
+
+            // cluster vaults collaterals must have the LTVs set in range with LTV separation provided
+            (uint16 borrowLTV, uint16 liquidationLTV,,,) = IEVault(vault).LTVFull(collateral);
+            testProperty(borrowLTV != liquidationLTV, ERROR__LTV_CONFIG);
+            testProperty(borrowLTV > 0 && borrowLTV <= 0.85e4, ERROR__LTV_CONFIG);
+            testProperty(liquidationLTV > 0 && liquidationLTV <= 0.9e4, ERROR__LTV_CONFIG);
+
+            // iterate over recognized collateral perspectives to check if the collateral is recognized
+            bool recognized = false;
+            for (uint256 j = 0; j < recognizedCollateralPerspectives.length; ++j) {
```

Comment:
If a vault has already been verified by a perspective, calling `perspectiveVerify` on it will almost immediately return without full verification. Or maybe I don't understand what you mean?

Reply:
Say you have 2 perspectives to check, and it's already been verified in the second one. Right now it will try the expensive full verification in the first one before even going to the second. I'm just wondering if we had 2 loops, one that first did the cheap check on each before falling back to the expensive ones.